core/src/main/java/com/novoda/rxpresso/mock/RxMocks.java

```diff
@@ -0,0 +1,240 @@
+        final String keyForArgs = getKeyFor(method, args);
+        final Observable observable = subject
+                .dematerialize()
+                .doOnEach(new NotifyDataEvent(notificationSubject))
+                .lift(new SwallowUnsubscribe());
+        observableHashMap.put(keyForArgs, observable);
+        mapSubject.put(observable, new Pair<>(subject, notificationSubject));
+    }
+
+    private String getKeyFor(Method method, Object[] args) {
+        StringBuilder keyBuilder = new StringBuilder(method.getName());
+        int index = 0;
+        for (Class<?> type : method.getParameterTypes()) {
+            keyBuilder.append('#').append(type.getSimpleName()).append('-').append(args[index++].hashCode());
+        }
+        return keyBuilder.toString();
+    }
+
+    private AddUnsubscribe clearOnUnsubscribe(final Object observable) {
+        return new AddUnsubscribe(
+                BooleanSubscription.create(
+                        new Action0() {
+                            @Override
+                            public void call() {
+                                mapSubject.get(observable).first.clear();
+                            }
+                        }
+                )
+        );
+    }
+
+    private static class NotifyDataEvent<T> implements Action1<Notification<? super T>> {
+
+        private final PublishSubject<Notification<T>> publishSubject;
+
+        public NotifyDataEvent(PublishSubject<Notification<T>> publishSubject) {
+            this.publishSubject = publishSubject;
+        }
+
+        @Override
+        public void call(Notification<? super T> notification) {
+            publishSubject.onNext((Notification<T>) notification);
+        }
+    }
+
+    private static class SwallowUnsubscribe<T> implements Observable.Operator<T, T> {
+
+        @Override
+        public Subscriber<? super T> call(final Subscriber<? super T> subscriber) {
+            return new Subscriber<T>() {
+                @Override
+                public void onCompleted() {
+                    subscriber.onCompleted();
+                }
+
+                @Override
+                public void onError(Throwable e) {
+                    subscriber.onError(e);
+                }
+
+                @Override
+                public void onNext(T t) {
+                    subscriber.onNext(t);
+                }
+            };
+        }
+
+    }
+
+    private static class AddUnsubscribe<T> implements Observable.Operator<T, T> {
+
+        private final Subscription unsubscribe;
+
+        private AddUnsubscribe(Subscription unsubscribe) {
+            this.unsubscribe = unsubscribe;
+        }
+
+        @Override
+        public Subscriber<? super T> call(final Subscriber<? super T> subscriber) {
+            subscriber.add(unsubscribe);
+            return subscriber;
+        }
+
+    }
+
+    private static Func2<Notification, Notification, Notification> unzip() {
```

Comment:
what does this mean?

The name zip is known when we're combining observables, but I don't follow how the given function here does the opposite of that - I see it's using Notifications and not Observables: is there some fact about functions that take notifications that could help me understand what unzip means in this context? without context, it looks like a function that could easily be named `ignoreFirst()`